File: src/main/java/com/ninja/ieumnori/ieumnoriserver/magazine/domain/MagazineCategory.java
package com.ninja.ieumnori.ieumnoriserver.magazine.domain;

import jakarta.persistence.*;
import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@Entity
@Table(name = "magazine_category")
public class MagazineCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    @Column(name = "is_deleted", nullable = false)
    private boolean isDeleted = false;

    @Column(name = "title", nullable = false)
    private String title;

    public void updateTitle(String newTitle) {
        this.title = newTitle;
    }

    public void markDeleted() {
        this.isDeleted = true;
    }
}

File: src/main/java/com/ninja/ieumnori/ieumnoriserver/magazine/domain/MagazineCategory.kt
package com.ninja.ieumnori.ieumnoriserver.magazine.domain

import com.ninja.ieumnori.ieumnoriserver.common.domain.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "magazine_category")
class MagazineCategory (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @Column(name = "category_name", nullable = false)
    var categoryName: String,
): BaseEntity() {
    constructor(): this(0, "")
}